### PR TITLE
[HttpKernel] Check if "symfony/proxy-manager-bridge" package is installed

### PR DESCRIPTION
**Pull request by hason**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -662,7 +662,7 @@ protected function getContainerBuilder()
     {
         $container = new ContainerBuilder(new ParameterBag($this->getKernelParameters()));
 
-        if (class_exists('ProxyManager\Configuration')) {
+        if (class_exists('ProxyManager\Configuration') && class_exists('Symfony\Bridge\ProxyManager\LazyProxy\Instantiator\RuntimeInstantiator')) {
```
Comment (aitboudad) [inline src/Symfony/Component/HttpKernel/Kernel.php]: you can remove the first condition

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 2]
```diff
@@ -682,7 +682,7 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
         // cache the container
         $dumper = new PhpDumper($container);
 
-        if (class_exists('ProxyManager\Configuration')) {
+        if (class_exists('ProxyManager\Configuration') && class_exists('Symfony\Bridge\ProxyManager\LazyProxy\PhpDumper\ProxyDumper')) {
```
Comment (aitboudad) [inline src/Symfony/Component/HttpKernel/Kernel.php]: same here

File: src/Symfony/Component/HttpKernel/Kernel.php (hunk 1 above)
Comment (hason) [inline src/Symfony/Component/HttpKernel/Kernel.php]: I think not. This PR solves a case when "symfony/framework-bundle" and "ocramius/proxy-manager" are installed. The first condition solves a case when "symfony/symfony" is installed and "ocramius/proxy-manager" not.

Comment (aitboudad) [inline src/Symfony/Component/HttpKernel/Kernel.php]: but I'm not sure If that makes any sense, if class ```RuntimeInstantiator``` exist then ```ProxyManager\Configuration``` exist also https://github.com/hason/symfony/blob/pm/src/Symfony/Bridge/ProxyManager/composer.json#L21 without checking it

Comment (stof) [inline src/Symfony/Component/HttpKernel/Kernel.php]: @aitboudad you are only looking at the case of installing subtree packages. When installing the fullstack repo, the RuntimeInstantiator class **always** exists, as it is part of the fullstack repo. But this does not mean that the ProxyManager library is installed

Comment (aitboudad) [inline src/Symfony/Component/HttpKernel/Kernel.php]: now I understand :)

**Comment by hason** [comment]
ping @aitboudad @fabpot

**Comment by aitboudad** [comment]
:+1:

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @hason.

**Pull request merged**